### PR TITLE
 YDBD binary args: added database quotas limits

### PR DESCRIPTION
**Pull request by Iliamish**

### CODE REVIEW

File: ydb/core/driver_lib/cli_utils/cli_cmds_tenant.cpp [hunk 1]
```diff
@@ -606,6 +620,40 @@ class TClientCommandTenantAddPools
...
+public:
+    TClientCommandTenantChangeQuotas()
+        : TTenantClientGRpcCommand("change", {}, "Change data size quotas for database")
+    {}
+
+    void Config(TConfig& config) override {
+        TTenantClientGRpcCommand::Config(config);
+        config.SetFreeArgsMin(0);
+        config.Opts->AddLongOption("data-size-hard-quota", "A maximum data size in bytes, new data will be rejected when exceeded")
+            .RequiredArgument("NUM").StoreResult(&DataSizeHardQuota);
```
Comment (Gazizonoki) [inline ydb/core/driver_lib/cli_utils/cli_cmds_tenant.cpp:641]: All options are optional by default. If you want to make it mandatory, use the Required() method.

File: ydb/core/driver_lib/cli_utils/cli_cmds_tenant.cpp [hunk 2]
```diff
@@ -606,6 +620,40 @@ class TClientCommandTenantAddPools
...
+        config.Opts->AddLongOption("data-size-soft-quota", "Data size in bytes (lower than data_size_hard_quota), at this value new data ingestion is re-enabled again")
+            .OptionalArgument("NUM").StoreResult(&DataSizeSoftQuota);
```
Comment (Gazizonoki) [inline ydb/core/driver_lib/cli_utils/cli_cmds_tenant.cpp:643]: Here, it's better to use RequiredArgument().

File: ydb/core/driver_lib/cli_utils/cli_cmds_tenant.cpp [hunk 3]
```diff
@@ -627,6 +675,15 @@ class TClientCommandTenantPools : public TClientCommandTree {
     }
 };
 
+class TClientCommandTenantQuotas : public TClientCommandTree {
+public:
+    TClientCommandTenantQuotas()
+        : TClientCommandTree("quotas", {}, "")
```
Comment (Gazizonoki) [inline ydb/core/driver_lib/cli_utils/cli_cmds_tenant.cpp:681]: You forgot to write the command description as the third argument. It's needed for displaying in --help.

**Review by Gazizonoki — CHANGES_REQUESTED** [review]

File: ydb/core/driver_lib/cli_utils/cli_cmds_tenant.cpp (hunk 3 above)
Comment (Iliamish) [reply ydb/core/driver_lib/cli_utils/cli_cmds_tenant.cpp:681]: I did the same here as in [`units`](https://github.com/ydb-platform/ydb/blob/f1b17cefba75c57c4c3f804599db55a19318a00a/ydb/core/driver_lib/cli_utils/cli_cmds_tenant.cpp#L660) and [`pools`](https://github.com/ydb-platform/ydb/blob/f1b17cefba75c57c4c3f804599db55a19318a00a/ydb/core/driver_lib/cli_utils/cli_cmds_tenant.cpp#L672) commands:

Is it better to add a description?

Comment (Gazizonoki) [reply ydb/core/driver_lib/cli_utils/cli_cmds_tenant.cpp:681]: But [create](https://github.com/ydb-platform/ydb/blob/main/ydb/core/driver_lib/cli_utils/cli_cmds_tenant.cpp#L311), [remove](https://github.com/ydb-platform/ydb/blob/main/ydb/core/driver_lib/cli_utils/cli_cmds_tenant.cpp#L382) and [status](https://github.com/ydb-platform/ydb/blob/main/ydb/core/driver_lib/cli_utils/cli_cmds_tenant.cpp#L239) commands have description. It is better to write description in quotas cmd

File: ydb/core/driver_lib/cli_utils/cli_cmds_tenant.cpp [hunk 4]
```diff
@@ -606,6 +620,40 @@ class TClientCommandTenantAddPools
...
+            .RequiredArgument("NUM").StoreResult(&DataSizeHardQuota);
+        config.Opts->AddLongOption("data-size-soft-quota", "Data size in bytes (lower than data_size_hard_quota), at this value new data ingestion is re-enabled again")
+            .RequiredArgument("NUM").StoreResult(&DataSizeSoftQuota);
+    }
+
+    void Parse(TConfig& config) override
+    {
+        TTenantClientGRpcCommand::Parse(config);
+
+        GRpcRequest.mutable_database_quotas()->set_data_size_hard_quota(DataSizeHardQuota);
```
Comment (Gazizonoki) [inline ydb/core/driver_lib/cli_utils/cli_cmds_tenant.cpp:650]: data-size-hard-quota option may be not specified, better to check it in ParseResult, like below

**Review by Gazizonoki — CHANGES_REQUESTED** [review]

**Review by Gazizonoki — DISMISSED** [review]

**Comment by Iliamish** [comment]
@Gazizonoki  Hi! The failed tests do not look like they are related to my request, but because of them i can't merge request. Can you take a look?

**Comment by Gazizonoki** [comment]
@Iliamish We still need to discuss whether we need this feature or not. I am not entitled to make this decision alone. We will come back with an answer a bit later.

**Review by CyberROFL — CHANGES_REQUESTED** [review]
Please get rid of the copy paste in:
- parser configuration;
- parsing.

**Comment by Iliamish** [comment]
> Please get rid of the copy paste in:
> 
> * parser configuration;
> * parsing.

Good afternoon! I don't quite understand where you can get rid of the copy paste in this code.

**Comment by Iliamish** [comment]
@CyberROFL Hi! Can you please come back to PR

**Comment by Iliamish** [comment]
@Gazizonoki, Hi! Are there any updates about this PR?

**Comment by Iliamish** [comment]
@CyberROFL Hi! Can you please review updates

**Review by CyberROFL — APPROVED** [review]

**Comment by CyberROFL** [comment]
@Iliamish please rebase

**Comment by Iliamish** [comment]
> @Iliamish Could you rebase this change?

done

**Pull request merged**